src/Components/Modal/modal.js

```diff
@@ -1,24 +1,100 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import "./modal.css";
 import ModalItems from "./Modal-Items/ModalItems";
 
 const Modal = () => {
-	return (
-		<div className="modal-class">
-			<p>Available Routes</p>
-			<span>
-				<label>Direction</label>
-				<label>Duration</label>
-				<label>Vehicle</label>
-				<label>Cost</label>
-			</span>
-			<ModalItems />
-			<ModalItems />
-			<ModalItems />
-			<ModalItems />
-			<ModalItems />
-		</div>
-	);
+	const [routes, setRoutes] = useState([]);
+
+	const fetchRoutes = () => {
+		let requestBody = {
+			query: `
+         query{
+            routes{
+              id
+              direction
+              vehicle
+              duration
+              cost
+            }
+       }
+      `,
+		};
+
+		fetch("https://ride-booking-app.herokuapp.com/graphql", {
```

Comment:
can you create a helper function that will handle this fetching? It's not necessary in this component

Reply:
you can add that to src/utils/index.js

```js
function makeRequest({url = REACT_APP_API_URL, method='post', data}) {
  return fetch(url, { 
      method,
      body: JSON.stringify(data),
      headers:   {
           "Content-Type": "application/json"
       }
    })
    .then(handleResponse)
    .catch(handleError)
}

// create the function for **handleResponse** and **handleError**
// Add REACT_APP_API_URL=https://ride-booking-app.herokuapp.com/graphql to your environment variable( **.env**)

Then any where you need to make call like here
makeRequest({ data: requestedBody }).....
```